ethereumj-core/src/main/java/org/ethereum/core/Transaction.java

```diff
@@ -253,9 +249,11 @@ public ECKey getKey() {
     }
 
     public void sign(byte[] privKeyBytes) throws MissingPrivateKeyException {
-        byte[] hash = this.getRawHash();
-        ECKey key = ECKey.fromPrivate(privKeyBytes).decompress();
-        this.signature = key.sign(hash);
+        sign(ECKey.fromPrivate(privKeyBytes));
```

Comment:
note:
I think `Transaction#sign(byte[])` should be deprecated and removed in the long-run.